tests/trainer/loops/test_evaluation_loop.py

```diff
@@ -65,3 +69,49 @@ def on_validation_end(self):
     trainer.fit(LessBoringModel())
 
     assert order == ["log_epoch_metrics", "on_validation_end"]
+
+
+@RunIf(min_gpus=1)
+def test_memory_consumption_validation(tmpdir):
+    """Test that the training batch is no longer in GPU memory when running validation"""
+
+    class BoringLargeBatchModel(BoringModel):
+
+        @property
+        def num_params(self):
+            num_params = 0
+            for params in self.parameters():
+                num_params += params.numel()
+            return num_params
+
+        def train_dataloader(self):
+            # batch target memory >= 100x boring_model size
+            batch_size = self.num_params * 100 // 32 + 1
+            return DataLoader(RandomDataset(32, 5000), batch_size=batch_size)
+
+        def val_dataloader(self):
+            # batch target memory >= 100x boring_model size
+            batch_size = self.num_params * 100 // 32 + 1
+            return DataLoader(RandomDataset(32, 5000), batch_size=batch_size)
+
+        def training_step(self, batch, batch_idx):
+            # there is a batch and the boring model, but not two batches on gpu, assume 32 bit = 4 bytes
+            assert 101 * self.num_params * 4 < torch.cuda.memory_allocated(
```

Comment:
```suggestion
            lower_bound < torch.cuda.memory_allocated() < upper_bound
```
I suggest to define upper and lower bound variables to bring the assert statement on one line.

Reply:
Done!